### PR TITLE
Improve goto definition for MBE

### PR DESCRIPTION
**Pull request by edwin0cheng**
This PR improve the macro resolution for goto definition and expression macro invocation by using proper path resolution for external macros.

### CODE REVIEW

File: crates/ra_ide_api/src/goto_definition.rs [hunk 1]
```diff
@@ -216,6 +216,29 @@ mod tests {
...
+        covers!(goto_definition_works_for_macros);
+        check_goto(
+            "
+            //- /lib.rs
+            use foo::foo;
+            fn bar() {
+                <|>foo!();
+            }
+
+            //- /foo/lib.rs
```
Comment (matklad) [inline crates/ra_ide_api/src/goto_definition.rs]: Hm, do we ever specify that `/foo/lib.rs` is an extern crate?

Comment (edwin0cheng) [reply crates/ra_ide_api/src/goto_definition.rs]: Yes if i understand correctly :)

https://github.com/rust-analyzer/rust-analyzer/blob/7a1cae59acf72f821343b2ba10ef69fb92a5b952/crates/ra_ide_api/src/mock_analysis.rs#L80-L100

**Comment by edwin0cheng** [comment]
bors retry

**Pull request merged**